editor/src/document/document_file.rs

```diff
@@ -372,12 +364,31 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 				Err(e) => log::error!("DocumentError: {:?}", e),
 				Ok(_) => (),
 			},
-			RenderDocument => responses.push_back(
-				FrontendMessage::UpdateCanvas {
-					document: self.document.render_root(),
-				}
-				.into(),
-			),
+			RenderDocument => {
+				responses.push_back(
+					FrontendMessage::UpdateCanvas {
+						document: self.document.render_root(),
+					}
+					.into(),
+				);
+				let viewport = ipp.viewport_size.as_f64();
+				let [bounds1, bounds2] = self.document.visible_layers_bounding_box().unwrap_or_default();
+				let bounds1 = bounds1.min(DVec2::ZERO) - viewport * 0.5;
+				let bounds2 = bounds2.max(viewport) + viewport * 0.5;
```

Comment:
could you try `out 2^(-¹/₃ • x ) * SCROLLBAR_PERCENT_AT_100_ZOOM (0.5 for now)` instead of 0.5?

Reply:
actually probably 1- that whole thing now that I think about it

Reply:
Done (sorry for late reply)